src/AsertoClient.ts

```diff
@@ -0,0 +1,71 @@
+import { post } from './api'
+
+const VISIBLE = 4
+const ENABLED = 2
+const ALLOWED = 1
+
+export { VISIBLE, ENABLED, ALLOWED }
+
+export interface ClientOptions {
+  accessToken: string
+  displayStateMapEndpoint?: string
+  serviceUrl?: URL
+  policyRoot: string
+}
+
+export default class AsertoClient {
+  token: string
+  endpoint: string
+  service: URL
+  policyRoot: string
+  private __displayStateMap: Record<string, Record<string, boolean>>
+
+  constructor(options?: ClientOptions) {
+    if (!options || !options.accessToken) {
+      throw new Error('AsertoClient: must provide access token')
+    }
+    this.token = options.accessToken
+    this.endpoint = options.displayStateMapEndpoint || '/__displaystatemap'
+    this.service = options.serviceUrl || new URL(window.location.origin)
+    this.policyRoot = options.policyRoot
+    this.__displayStateMap = {}
+  }
+
+  async reload(body: RequestInit['body'], headers: Record<string, string> = {}) {
```

Comment:
On typing the body as RequestInit['body'] - this means that the callers (including @aserto/aserto-react) will need to type it that way as well.  I think this PR may not currently work with the current impl of aserto-dev/aserto-react, which has its own type defs (and types headers as any and body as any).

Reply:
But presumably we would want `aserto-react` to import the types that we export from `aserto-spa-js`?

Reply:
We can think of `aserto-spa-js` as being like any dependency of `aserto-react`, an implementation detail. If `aserto-spa-js` changes the typing of its API, it doesn't affect the typing of `aserto-react`s API unless `aserto-react` explicitly derives its types from `aserto-spa-js` (which it doesn't, since it explicitly types `body` itself).

So with this change `aserto-spa-js` will start rejecting invalid `body` parameters, but that doesn't affect users of `aserto-react` at all.  It would be preferable for `aserto-react` to refine its typing too, but that's something we can easily do as a follow-up to this.